### PR TITLE
🥛 Make app navigation semi-transparent

### LINKED ISSUE #3047: 🥛 Make app navigation semi-transparent
As per the mockups

Bildschirmfoto vom 2022-08-16 12-58-35

the app navigation component should become semi-transparent and let the background image show through with a blur.

## Specification

- 80% opaque white with 10px blur, like in Dashboard. Code: `background-color: var(--color-background-translucent); -webkit-backdrop-filter: var(--background-blur); backdrop-filter: var(--background-blur);`
- [AppNavigationNew](https://github.com/nextcloud/nextcloud-vue/blob/master/src/components/AppNavigationNew/AppNavigationNew.vue) gets our Nextcloud fade, looks nice and clicky
- WCAG AA: Contrast of secondary text in the navigation works after darkening it to `5c5c5c`
- WCAG AAA: Just like for Dashboard, the high contrast theme should make the navigation background fully opaque and the secondary text color also color-main-text.

### PR DESCRIPTION
**Pull request by GretaD**
- [x] requires https://github.com/nextcloud/server/pull/33610

fixes #3047 

- [x] the image is just for testing, it will be added globally on server. I will remove it after review

appnaaav

### CODE REVIEW

File: src/components/Content/Content.vue [hunk 1]
```diff
@@ -85,6 +85,8 @@ export default {
 	display: flex;
 	padding-top: 50px;
 	min-height: 100%;
+	background: url("/src/assets/kamil-porembinski-clouds.jpg");
```
Comment (ChristophWurst) [inline src/components/Content/Content.vue:88]: to be removed once server puts the image behind every app content page

**Review by ChristophWurst — COMMENTED** [review]
Looks good so far

File: styleguide/assets/variables.css [hunk 2]
```diff
@@ -37,6 +37,8 @@
 	--color-box-shadow: rgba(77, 77, 77, 0.5);
 	--color-border: #ededed;
 	--color-border-dark: #dbdbdb;
+	--color-main-background-semitransparent: rgba(255,255,255,0.8);
+	--background-blur: blur(10px);
```
Comment (jancborchardt) [inline styleguide/assets/variables.css:41]: Let’s go with slightly higher, as per @marcoambrosini feedback :)
```suggestion
	--background-blur: blur(25px);
```

**Comment by GretaD** [comment]
the variables names will be changes because i didnt see julius pr: https://github.com/nextcloud/server/pull/33610/files

but doesnt change the result

File: styleguide/assets/variables.css [hunk 3]
```diff
@@ -37,6 +37,8 @@
 	--color-box-shadow: rgba(77, 77, 77, 0.5);
 	--color-border: #ededed;
 	--color-border-dark: #dbdbdb;
+	--color-main-background-semitransparent: rgba(255,255,255,0.8);
```
Comment (jancborchardt) [inline styleguide/assets/variables.css:40]: We already have a variable for this actually, and it’s used in Dashboard: `var(--color-background-translucent)`

Comment (jancborchardt) [reply styleguide/assets/variables.css:40]: This will also ensure it works with dark mode.

File: styleguide/assets/variables.css (hunk 2 above)
Comment (GretaD) [reply styleguide/assets/variables.css:41]: yes, Julius added new variables https://github.com/nextcloud/server/pull/33610/files and i will use his :)

**Review by jancborchardt — DISMISSED** [review]
See comments :)

File: styleguide/assets/variables.css (hunk 3 above)
Comment (GretaD) [reply styleguide/assets/variables.css:40]: this is just for documentation, we get the variables from theming now

**Comment by jancborchardt** [comment]
Ah ok got it, then just use @juliushaertl’s variables, my comments are moot. :)

File: src/components/Content/Content.vue (hunk 1 above)
Comment (juliushaertl) [reply src/components/Content/Content.vue:88]: Let me push that to my server pr as well then

Comment (juliushaertl) [reply src/components/Content/Content.vue:88]: Available now as --image-main-background which will also adapt to theming https://github.com/nextcloud/server/pull/33610

**Comment by ChristophWurst** [comment]
> requires [nextcloud/server#33610](https://github.com/nextcloud/server/pull/33610)

What are we going to do in terms of server compatibility? Is this a hard requirement? What happens in the absence of the new variables and/or the background image?

1) Declare the next major of this lib to only work with 25+
2) Make the component work with and without the vars/image in server

**Review by marcoambrosini — COMMENTED** [review]
this border is not needed anymore https://github.com/nextcloud/nextcloud-vue/blob/0d73e367f29fdad1461e4d2bea6e216342119ddc/src/components/AppNavigation/AppNavigation.vue#L158

File: src/components/NcAppNavigation/NcAppNavigation.vue [hunk 4]
```diff
@@ -151,17 +151,17 @@ export default {
...
 	user-select: none;
-	border-right: 1px solid var(--color-border);
 	display: flex;
 	flex-direction: column;
 	flex-grow: 0;
 	flex-shrink: 0;
-
+	background-color:  var(--color-main-background-blur);
+	-webkit-backdrop-filter: var(--filter-background-blur);
+	backdrop-filter: var(--filter-background-blur);
```
Comment (juliushaertl) [inline src/components/NcAppNavigation/NcAppNavigation.vue:164]:
```suggestion
	background-color:  var(--color-main-background-blur, var(--color-main-background));
	-webkit-backdrop-filter: var(--filter-background-blur, none);
	backdrop-filter: var(--filter-background-blur, none);
```

To make it a non-breaking change and still usable on older Nextcloud versions.

File: src/components/NcContent/NcContent.vue [hunk 5]
```diff
@@ -85,6 +85,8 @@ export default {
 	display: flex;
 	padding-top: 50px;
 	min-height: 100%;
+	background: url("/src/assets/kamil-porembinski-clouds.jpg");
+	background-size: contain;
```
Comment (juliushaertl) [inline src/components/NcContent/NcContent.vue:89]: Let's remove this as it will be added with #3091

**Review by ChristophWurst — APPROVED** [review]
:+1: code looks good

**Review by juliushaertl — APPROVED** [review]

**Comment by raimund-schluessler** [comment]
@juliushaertl So this will work with NC 24 and below as well, right?

**Comment by juliushaertl** [comment]
Yes, as we have the fallback to the previous values in place.

**Pull request merged**